### ISSUE TITLE
Support finding paths between modules

### ISSUE THREAD

**Issue by mjpitz** [depscloud/depscloud#12]
Given two nodes in the graph (start and end), output all paths that exist from start to end. This should probably be implemented as a client-side feature (in the CLI or in the API). The result should be a 2 dimensional array of Module. Each entry in the outermost array would be a path. Each path would be an array of modules.

**Comment by MadhuvanthG** [comment]
@mjpitz, I'm interested in having a look at this. I think, we would be able to leverage our BFS / DFS implementation. Something like this-https://www.geeksforgeeks.org/find-paths-given-source-destination/?

**Comment by MadhuvanthG** [comment]
Also, just curious.. What are some valid use-cases that would need this data?

**Comment by mjpitz** [comment]
Yup. I'd use DFS to enumerate paths. Path enumeration can be valuable for a few different reasons. For dependency management, you might want to know how a given library is pulled into an application. Another example would be if a CVE is discovered in a library (at some version), and you want to know if / how it affects a specific application (or set of applications).
